### PR TITLE
Refactoring python client.

### PR DESCRIPTION
**Pull request by alainjobart**
@dumbunny @aaijazi similar to internal fixes.

---
This change is [Reviewable](https://reviewable.io/reviews/youtube/vitess/1578)

### CODE REVIEW

File: py/vtdb/proto3_encoding.py [hunk 1]
```diff
@@ -259,33 +275,32 @@ def begin_request(self, effective_caller_id):
...
-    Args:
-      effective_caller_id: optional vtgate_client.CallerID.
+    Uses the effective_caller_id saved from begin_request().
 
     Returns:
       A vtgate_pb2.RollbackRequest object.
     """
     request = vtgate_pb2.RollbackRequest()
-    self._add_caller_id(request, effective_caller_id)
+    self._add_caller_id(request, self._effective_caller_id)
```
Comment (dumbunny) [inline py/vtdb/proto3_encoding.py]: Should you clear self._effective_caller_id in rollback and commit?

Comment (alainjobart) [inline py/vtdb/proto3_encoding.py]: I agree I will add it.

**Comment by alainjobart** [comment]
@dumbunny PTAL this is ready to go I think. Upon import I'll have to fix stubbyp3 accordingly.

**Comment by dumbunny** [comment]
LGTM

**Pull request merged**